DXMainClient/Domain/Multiplayer/CnCNet/CnCNetAnnouncement.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Rampastring.Tools;
+using System;
+using System.IO;
+using System.Net;
+
+namespace DTAClient.Domain.Multiplayer.CnCNet
+{
+    /// <summary>
+    /// Announcement controlled by a remote url for the CnCNet Lobby
+    /// </summary>
+    public class CnCNetAnnouncement
+    {
+        public CnCNetAnnouncement()
+        {
+            Color = ClientCore.ClientConfiguration.Instance.CnCNetAnnouncementColorRGB.Split(',');
+        }
+
+        public string GetAnnouncementMessage()
+        {
+            string message = "";
+
+            try
+            {
+                WebClient client = new WebClient();
+
+                Stream response = client.OpenRead(ClientCore.ClientConfiguration.Instance.CnCNetAnnouncementURL);
+                using (StreamReader reader = new StreamReader(response))
+                {
+                    message = reader.ReadToEnd();
+                }
+                return message;
+            }
+            catch(Exception ex)
+            {
+                Logger.Log("Error fetching Announcement text from remote url: " 
+                    + ClientCore.ClientConfiguration.Instance.CnCNetAnnouncementURL 
+                    + " Error: " + ex.Message);
+            }
+
+            return message;
+        }
+
+        public string[] Color { get; private set; }
```

Comment:
This string array could be replaced with Microsoft.Xna.Framework.Color. Then you could use
AssetLoader.GetColorFromString for parsing the color.

You could also parse the color from the announcement by including a separator character between the actual announcement and the color data. Doing this instead of coding the color locally would make the feature much more flexible.

Reply:
This also needs to be multi-threaded, because this can significantly increase client load time due to the web request during startup. Multi-threading is hard to do right though, so I could make that change if you're not very familiar with it.